src/pages/ImportData.js

```diff
@@ -60,16 +61,29 @@ const ChoiceButton = ({ iconName, title, detail, style, ...props }) => {
 const ImportData = () => {
   const { workspaces, refresh: refreshWorkspaces, loading: loadingWorkspaces } = useWorkspaces()
   const [isImporting, setIsImporting] = useState(false)
-  const { query: { url, format, ad, wid, template, snapshotId, snapshotName } } = Nav.useRoute()
+  const { query: { url, format, ad, wid, template, snapshotId, snapshotName, snapshotIds, referrer } } = Nav.useRoute()
   const [mode, setMode] = useState(wid ? 'existing' : undefined)
   const [isCreateOpen, setIsCreateOpen] = useState(false)
   const [isCloneOpen, setIsCloneOpen] = useState(false)
   const [selectedWorkspaceId, setSelectedWorkspaceId] = useState(wid)
   const [selectedTemplateWorkspaceKey, setSelectedTemplateWorkspaceKey] = useState()
   const [allTemplates, setAllTemplates] = useState()
 
+  const { dataCatalog } = useDataCatalog()
+  const snapshots = _.flow(
+    _.filter(snapshot => _.includes(snapshot['dct:identifier'], snapshotIds)),
+    _.map(snapshot => ({ id: snapshot['dct:identifier'], title: snapshot['dct:title'] }))
+  )(dataCatalog)
+
   const isDataset = format !== 'snapshot'
   const noteMessage = 'Note that the import process may take some time after you are redirected into your destination workspace.'
+  let title = `Import ${isDataset ? 'Data' : 'Snapshot'}`
+  let header = `Importing ${isDataset ? 'Data' : `Snapshot ${snapshotName}`}`
+
+  if (referrer === 'data-catalog') {
+    title = 'Catalog'
+    header = 'Linking data to a workspace'
+  }
```

Comment:
Would this work? It will prevent the need for mutation
```suggestion
  const [title, header] = Utils.cond(
    [referrer === 'data-catalog', () => ['Catalog', 'Linking data to a workspace']],
    [isDataset, () => ['Data', 'Data']],
    [Utils.DEFAULT, () => ['Snapshot', `Snapshot ${snapshotName}`]]
  )
```